### ISSUE TITLE
Strange version resolving of nested dependencies

### ISSUE THREAD

**Issue by Lebedancer** [yarnpkg/yarn#4034]
**Do you want to request a *feature* or report a *bug*?**
maybe bug

**What is the current behavior?**
wrong nested dependencies package version resolving

**If the current behavior is a bug, please provide the steps to reproduce.**
I have 2 dependencies in my repo 

```
"dependencies": {
    "jquery-form": "^4.2.1",
    "jquery": "2.1.4"
  }
```

jquery-form has own dependency
```
"dependencies": {
     "jquery": ">=1.7"
  }
```

when i use yarn install, i see this in lock file:

```
jquery@2.1.4:
  version "2.1.4"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-2.1.4.tgz#228bde698a0c61431dc2630a6a154f15890d2317"

jquery@>=1.7:
  version "3.2.1"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-3.2.1.tgz#5c4d9de652af6cd0a770154a631bba12b015c787"
```

**What is the expected behavior?**
```
jquery@2.1.4, jquery@>=1.7:
  version "2.1.4"
  resolved "https://registry.yarnpkg.com/jquery/-/jquery-2.1.4.tgz#228bde698a0c61431dc2630a6a154f15890d2317"
```

**Please mention your node.js, yarn and operating system version.**
Windows -  10,
Node -  6.9.1
Yarn -  0.27.5

**Comment by BYK** [comment]
Duplicate of #4024. Please read the discussion there.

**Comment by Lebedancer** [comment]
Are you sure? It looks like ```jquery@>=1.7``` doesn't matter result version it could be 1.9 / 2.1.4 / 3.2.1
But ```jquery@2.1.4``` wants exactly 2.1.4, so why should we have several versions of jquery, if we can have just one (2.1.4)?
Ps: sorry for my english :)

**Comment by BYK** [comment]
@Lebedancer - Please read the discussion I referred to :)

TL;DR
`jquery-form` needs to list `jquery` as a `peerDependency`. ;)
